lib/Driver/ToolChains.cpp

```diff
@@ -930,6 +931,8 @@ toolchains::Darwin::constructInvocation(const InterpretJobAction &job,
                                      runtimeLibraryPath);
   addPathEnvironmentVariableIfNeeded(II.ExtraEnvironment, "DYLD_FRAMEWORK_PATH",
                                      ":", options::OPT_F, context.Args);
+  addPathEnvironmentVariableIfNeeded(II.ExtraEnvironment, "DYLD_FRAMEWORK_PATH",
+                                     ":", options::OPT_Fsystem, context.Args);
```

Comment:
This won't combine them, unfortunately. I'm okay with this going unimplemented for now, though; please just file a JIRA bug about it.